src/odemis/util/test/transform_test.py

```diff
@@ -329,6 +330,47 @@ def _angle_diff(x, y):
     return min(y - x, y - x + 2.0 * numpy.pi, y - x - 2.0 * numpy.pi, key=abs)
 
 
+class ToPhysicalSpaceKnownValues(unittest.TestCase):
+
+    def setUp(self):
+        self._ji = [(0, 0), (0, 4), (7, 0), (7, 4)]
+        self._xy = [(-2, 3.5), (2, 3.5), (-2, -3.5), (2, -3.5)]
+        self._shape = (8, 5)
+
+    def test_to_physical_space_known_values(self):
+        """
+        to_physical_space should return known result with known input.
+
+        """
+        for ji, xy in zip(self._ji, self._xy):
+            res = to_physical_space(ji, self._shape)
+            numpy.testing.assert_array_almost_equal(xy, res)
+
+    def test_to_physical_space_multiple(self):
+        """
+        to_physical_space should return the same result when called on a list
+        of indices, or on individual indices.
+
+        """
+        _xy = to_physical_space(self._ji, self._shape)
+        for ji, xy in zip(self._ji, _xy):
+            res = to_physical_space(ji, self._shape)
+            numpy.testing.assert_array_almost_equal(xy, res)
```

Comment:
This test compares the values found for the list with the values found when calculated individually. If there is a systematic error in your method, you will not find it with this. You could argue that you test the method already in the previous test and check there the correct values are returned, but I would not rely on this. I think the test is stronger, if you also check here that the values returned match with the known values.

As you also allow an ndarray as input, I would suggest to also test the method for that type.

Reply:
This unit tests only tests that the result is the same if you call to_physical_space once on a list of tuples, or once for every element of that list. If the return values are wrong, the first unit test will fail, and this test case will still pass.

This is the intended behaviour. I believe a unit test should test one thing only. If after a code change the return values are wrong, I want only a single test case failing so that I know exactly what goes wrong.